keyboards/dz60/keymaps/bingocaller/keymap.c

```diff
@@ -0,0 +1,115 @@
+#include QMK_KEYBOARD_H
+
+enum custom_keycodes {
+    WORD_BACK = SAFE_RANGE,
+    WORD_FORWARD,
+    DELETE_WORD_BACK,
+    DELETE_WORD_FORWARD,
+    FINE_VOLUP,
+    FINE_VOLDOWN
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        switch (keycode) {
+            case WORD_BACK:
+                SEND_STRING(SS_DOWN(X_LALT) SS_TAP(X_LEFT) SS_UP(X_LALT));
+                return false;
+            case WORD_FORWARD:
+                SEND_STRING(SS_DOWN(X_LALT) SS_TAP(X_RIGHT) SS_UP(X_LALT));
+                return false;
+            case DELETE_WORD_BACK:
+                SEND_STRING(SS_DOWN(X_LALT) SS_TAP(X_BSPACE) SS_UP(X_LALT));
+                return false;
+            case DELETE_WORD_FORWARD:
+                SEND_STRING(SS_DOWN(X_LALT) SS_TAP(X_DELETE) SS_UP(X_LALT));
+                return false;
+            case FINE_VOLUP:
+                SEND_STRING(SS_DOWN(X_LALT) SS_DOWN(X_LSHIFT) SS_TAP(X__VOLUP) SS_UP(X_LALT) SS_UP(X_LSHIFT));
+                return false;
+            case FINE_VOLDOWN:
+                SEND_STRING(SS_DOWN(X_LALT) SS_DOWN(X_LSHIFT) SS_TAP(X__VOLDOWN) SS_UP(X_LALT) SS_UP(X_LSHIFT));
+                return false;
+            default:
+                return true;  // Process all other keycodes normally
+        }
+    }
+    return true;
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /*  Default layer:
+        * Space Cadet shifts (parentheses on tap)
+        * Caps Lock is Control on hold, Esc on tap
+        * Hyper/Caps Lock on Control
+        * Hold D to activate layer 1
+        * Hold Space to activate layer 3 (Mouse keys)
+        * Hold FN to activate layer 4
+    */
+    LAYOUT(
+        KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_NO, KC_BSPC,
+        KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS,
+        LCTL_T(KC_ESC), KC_A, KC_S, LT(1, KC_D), KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_ENT,
+        KC_LSPO, KC_NO, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSPC, KC_NO,
+        ALL_T(KC_CAPS), KC_LALT, KC_LGUI, LT(3, KC_SPC), LT(3, KC_SPC), LT(3, KC_SPC), KC_RGUI, KC_RALT, KC_NO, MO(3), ALL_T(KC_CAPS)),
+
+    /*  Layer 1:
+        * Vim arrows (HJKL)
+        * Vim-like move across words with W(ord), and B(eginning)
+        * Media controls (fine volume controls using Option+Shift)
+        * Backspace/Del on N/M
+        * Hold F to activate layer 2
+    */
+    LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
```

Comment:
It may be a good idea to change these `KC_TRNS`s to `_______` so it's easier to see which keys you've defined.

Reply:
That’s a good idea, I'll get right on it!